src/cargo/core/shell.rs

```diff
@@ -147,27 +147,61 @@ impl MultiShell {
 }
 
 impl Shell {
-    pub fn create(out: Box<Write + Send>, config: ShellConfig) -> Shell {
+    pub fn create(out: Box<Write + Send>, config: ShellConfig) -> CargoResult<Shell> {
```

Comment:
The problem there is, that the `out` is consumed - It'd needed to be newly created by the caller. As such I'm not really sure if that's at all possible